core/blockchain.go

```diff
@@ -2442,6 +2489,20 @@ func (bc *BlockChain) UpdateStakingMetaData(
 			batch, createValidator.ValidatorAddress, createValidator.ValidatorAddress, root,
 		)
 	case staking.DirectiveEditValidator:
+		editVal := decodePayload.(*staking.EditValidator)
+		if active := editVal.Active; active != nil && *active {
+			list, err := bc.ReadValidatorList()
+			if err != nil {
+				return err
+			}
+			if list == nil {
+				list = []common.Address{}
+			}
+			list = utils.AppendIfMissing(list, editVal.ValidatorAddress)
+			if err := bc.WriteValidatorList(batch, list); err != nil {
```

Comment:
No removal when it turns to inactive? Then this part of the code is always no-ops, because the ValidatorList contains all the validators.

Reply:
nvm, I see it later.